spec/java/util/HashSet.KeyIterator.lsl

```diff
@@ -89,12 +89,9 @@ automaton HashSet_KeyIteratorAutomaton
         if (!atValidPosition)
             action THROW_NEW("java.util.NoSuchElementException", []);
 
-        val key: Object = action SYMBOLIC("java.lang.Object");
-        action ASSUME(key != null);
-        action ASSUME(key != this.currentKey);
         val parentStorage: map<Object, Object> = HashSetAutomaton(this.parent).storage;
-        val sourceStorageHasKey: boolean = action MAP_HAS_KEY(parentStorage, key);
-        action ASSUME(sourceStorageHasKey);
+        val key: Object = action MAP_GET_ANY_KEY(parentStorage);
+        action ASSUME(key != this.currentKey);
```

Comment:
Consider using supplementary `unseen: map<Object, Object>` to store items for later retrieval as suggested by Valentin.

Reply:
Fixed